eclair-core/src/main/scala/fr/acinq/eclair/channel/fund/InteractiveTxBuilder.scala

```diff
@@ -622,19 +572,13 @@ private class InteractiveTxBuilder(replyTo: ActorRef[InteractiveTxBuilder.Respon
         case Left(cause) =>
           replyTo ! RemoteFailure(cause)
           unlockAndStop(session)
-        case Right((completeTx, fundingOutputIndex)) =>
-          stash.unstashAll(signCommitTx(completeTx, fundingOutputIndex))
+        case Right(completeTx) =>
+          signCommitTx(completeTx)
```

Comment:
Note that even though we stop the `InteractiveTxBuilder` after exchanging `tx_complete`, we still generate our `commit_sig` and `tx_signatures` before returning the `Succeeded` event. This way the channel state machine already has these messages available, and simply needs to send them at the right time. This greatly simplifies the channel state machine and makes it easier to reuse code for RBF and splicing.

I also factored the small state machine of the signature exchange into types (see `InteractiveTxSigningStatus` at the bottom of this file) instead of using an actor, let me know what you think about that design choice.